### PR TITLE
feat: added a build button that takes user directly to edit mode #1281

### PR DESCRIPTION
**Pull request by Priyanshu44**
image

Issue: 
-> https://github.com/awslabs/iot-application/issues/1277
-> https://github.com/awslabs/iot-application/issues/1281

Description: When a user creates a dashboard, they should be brought directly to the edit mode & Add a new "build" button next to the "view" button, which opens the dashboard directly in "edit" mode

Changes done in the CR:
-> User redirected to edit mode on creating dashboard.
-> Added a new build button, which opens the dashboard directly in edit mode.

### CODE REVIEW

**Review by mnischay — DISMISSED** [review]

**Comment by corteggiano** [comment]
please pull latest changes from main so that the tests+lints can run :)

**Review by mnischay — CHANGES_REQUESTED** [review]
Also, please rebase with the lastest rc branch

File: apps/client/src/hooks/dashboard/use-emit-viewmode.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { setIsEditModeVisibleAtom } from '~/store/view-mode';
+import { useSetAtom } from 'jotai';
+
+/** Use write-only view mode global store. */
+export function useEmitEditMode() {
```
[the remark this answers is not in the document]
Comment (Priyanshu44) [reply apps/client/src/hooks/dashboard/use-emit-viewmode.ts]: resolved

File: apps/client/src/hooks/dashboard/use-viewmode.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { isEditModeVisibleAtom } from '~/store/view-mode';
+import { useAtomValue } from 'jotai';
+
+/** Use readonly view mode global store. */
+export function useEditMode() {
+  return useAtomValue(isEditModeVisibleAtom);
```
[the remark this answers is not in the document]
Comment (Priyanshu44) [reply apps/client/src/hooks/dashboard/use-viewmode.ts]: resolved

**Comment by Priyanshu44** [comment]
Issue resolved

**Review by mnischay — APPROVED** [review]

**Comment by Priyanshu44** [comment]
Created another PR from main repo so closing this one

**Pull request closed without merging**